### PR TITLE
feat: contract caching

### PR DESCRIPTION
**Pull request by QuantumExplorer**
## Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: drive/src/drive/cache.rs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+use crate::contract::Contract;
+use crate::drive::flags::StorageFlags;
+use crate::fee::FeeResult;
+use costs::OperationCost;
+use moka::sync::Cache;
+use std::sync::Arc;
+
+/// Drive cache struct
+pub struct DriveCache {
```
Comment (shumkov) [inline drive/src/drive/cache.rs:6]: I think it's a bad idea to combine all caches together in this case you need to pass to the contract module the whole thing including other caches.

File: drive/src/drive/cache.rs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    /// Cached contracts
+    pub cached_contracts: Cache<[u8; 32], Arc<ContractFetchInfo>>,
+    /// Genesis time in ms
+    pub genesis_time_ms: Option<u64>,
+}
+
+/// Contract and fetch information
+pub struct ContractFetchInfo {
```
Comment (shumkov) [inline drive/src/drive/cache.rs]: According to the module hierarchal, this structure should be defined in the contract module where you place fetch functions.

**Review by shumkov — APPROVED** [review]

File: drive/src/drive/cache.rs (hunk 1 above)
Comment (QuantumExplorer) [reply drive/src/drive/cache.rs:6]: I just moved it to it's own file.

File: drive/src/drive/cache.rs (hunk 2 above)
Comment (QuantumExplorer) [reply drive/src/drive/cache.rs]: you are correct... will fix it.

**Pull request merged**